gobblin-runtime/src/main/java/org/apache/gobblin/runtime/TaskStateCollectorService.java

```diff
@@ -206,12 +222,53 @@ public Void call() throws Exception {
 
     LOGGER.info(String.format("Collected task state of %d completed tasks", taskStateQueue.size()));
 
+    Long taskByteSize;
+    String stringSize;
+    Double newPercentageCopied;
     // Add the TaskStates of completed tasks to the JobState so when the control
     // returns to the launcher, it sees the TaskStates of all completed tasks.
     for (TaskState taskState : taskStateQueue) {
       consumeTaskIssues(taskState);
       taskState.setJobState(this.jobState);
       this.jobState.addTaskState(taskState);
+
```

Comment:
This method is already quite complicated. Generally, we should try to have no more a screens of code in each method, to make it easier to understand and read. You can try moving the logic to a separate method or class, whatever makes more sense.

Fields like "lastPercentageReported" can be more meaningful in separate class, rather than in task state collector. That will better match single responsibility principle - https://en.wikipedia.org/wiki/SOLID

Reply:
Moving to a separate method for clarity because the state of the progress related class is tied to one instance of the taskCollectorService. If it still makes sense to separate the two into classes then I can attempt that.

Reply:
I think a separate method for now is good here, until we can determine more use cases/extensions to the progress reporting